interop-cats/shared/src/main/scala/zio/interop/catszmanaged.scala

```diff
@@ -211,3 +214,33 @@ private class CatsZManagedSync[R] extends CatsZManagedMonadError[R, Throwable] w
       }
     }
 }
+
+private object CatsZManagedArrowChoice extends ArrowChoice[ZManaged[*, Any, *]] {
+  final override def lift[A, B](f: A => B): ZManaged[A, Any, B] = ZManaged.fromFunction(f)
+  final override def compose[A, B, C](f: ZManaged[B, Any, C], g: ZManaged[A, Any, B]): ZManaged[A, Any, C] =
+    g >>= f.provide
+  final override def id[A]: ZManaged[A, Any, A] = ZManaged.environment
+  final override def dimap[A, B, C, D](fab: ZManaged[A, Any, B])(f: C => A)(g: B => D): ZManaged[C, Any, D] =
+    fab.provideSome(f).map(g)
+
+  final override def choose[A, B, C, D](
+    f: ZManaged[A, Any, C]
+  )(g: ZManaged[B, Any, D]): ZManaged[Either[A, B], Any, Either[C, D]] =
+    ZManaged.accessManaged(_.fold(f.provide(_).map(Left(_)), g.provide(_).map(Right(_))))
+  final override def first[A, B, C](fa: ZManaged[A, Any, B]): ZManaged[(A, C), Any, (B, C)] =
+    ZManaged.accessManaged { case (a, c) => fa.provide(a).map((_, c)) }
+  final override def second[A, B, C](fa: ZManaged[A, Any, B]): ZManaged[(C, A), Any, (C, B)] =
+    ZManaged.accessManaged { case (c, a) => fa.provide(a).map((c, _)) }
+  final override def split[A, B, C, D](f: ZManaged[A, Any, B], g: ZManaged[C, Any, D]): ZManaged[(A, C), Any, (B, D)] =
+    ZManaged.accessManaged { case (a, c) => f.provide(a).zip(g.provide(c)) }
+  final override def merge[A, B, C](f: ZManaged[A, Any, B], g: ZManaged[A, Any, C]): ZManaged[A, Any, (B, C)] = f.zip(g)
+
+  final override def lmap[A, B, C](fab: ZManaged[A, Any, B])(f: C => A): ZManaged[C, Any, B] = fab.provideSome(f)
+  final override def rmap[A, B, C](fab: ZManaged[A, Any, B])(f: B => C): ZManaged[A, Any, C] = fab.map(f)
+
+  final override def choice[A, B, C](f: ZManaged[A, Any, C], g: ZManaged[B, Any, C]): ZManaged[Either[A, B], Any, C] =
```

Comment:
I think you should be able to implement most of these by just calling the corresponding methods on `ZManaged`. For example `choice` is just `f ||| g`.

Reply:
I see, thanks, doing!

Reply:
done